### DISCUSSION TITLE
500.30 - ASP.NET Core app failed to start

### DISCUSSION THREAD

**Discussion by l0chte** [MCLD/greatreadingadventure#1100] [Help!]
Hello, I'm running Windows Server 2019 + SQL Server and following the manual instructions. After creating the databases (using script) and all installation steps, I have been troubleshooting the following error for about a week now to no avail:

**When loading localhost:**
HTTP Error 500.30 - ASP.NET Core app failed to start

**In the log:** 
2024-04-12 10:24:18.278 -04:00 [Information] GRA v"4.4.1" instance "n/a" environment "Production" in "C:\inetpub\wwwroot\wwwroot" with content root "C:\inetpub\wwwroot\"
2024-04-12 10:24:19.035 -04:00 [Information] Cache: in-memory
2024-04-12 10:24:19.069 -04:00 [Debug] Cache miss for "Startup": caching "string" (length 19) expires "00:00:10" ("absolute")
2024-04-12 10:24:22.958 -04:00 [Warning] Applying 54 database migrations, last is: "20230630231754_news-stand-update-sort"
2024-04-12 10:24:24.561 -04:00 [Error] Failed executing DbCommand ("20"ms) [Parameters=[""], CommandType='Text', CommandTimeout='60']"
""CREATE DATABASE [SRP];"
2024-04-12 10:24:24.591 -04:00 [Error] Startup error applying database migrations: "CREATE DATABASE permission denied in database 'master'."

Is there anywhere else I can look for error messages that might help narrow down the issue?

**Comment by k7hpn** [comment]
Hello!

So it looks to me, based on this log line:

> `[Error] Startup error applying database migrations: "CREATE DATABASE permission denied in database 'master'."`

That the software is trying to create the database. My guess is that it cannot access the database that you created with the script so it's trying to create its own. This leads me to think that there might be an issue with the connection string in your application configuration file. Check your `appsettings.json` file. You want to see something like this inside the main `{}` section of the file:

```json
  "ConnectionStrings": {
    "SqlServer": "Server=<server>;Database=<database>;user id=<user>;password=<password>;MultipleActiveResultSets=true;Encrypt=false",
  },
```

- `<server>` is the hostname of your server
- `<database>` is the name of the database (if you didn't change the script, it's `SRP`)
- `<user>` is the database user (if you changed it in the top of the SQL script, if you didn't change it the value is: `srp_owner`
- `<password>` is the database user password (which you changed in the SQL script)

At some point that `Encrypt=false` parameter seems to have become required but I'm not exactly sure when/where - if that's missing it's a good first thing to try.

Let me know if this helps!

**Comment by l0chte** [comment]
After weeks of messing around.... This morning I added db_owner permissions to the Builtin/Users group in SQL Server Management Studio and got it going...  Is this a security risk?

**Reply by k7hpn** [reply]
Hmm. The script should have [added `db_owner` as a role](https://github.com/MCLD/greatreadingadventure/blob/main/db/SQL-Server-createdb.sql#L230) on the `SRP` database to the user that it created. That change should make your change to the `Builtin/Users` group not necessary. That being said, I'm not exactly sure why your change made things work. The user that you created does require `db_owner` on the SRP database so that it can create and alter tables but shouldn't require (or have) that permission on other databases on your SQL Server.
